### PR TITLE
Kafka_7064 - bug introduced when switching config commands to ConfigResource 

### PR DESCRIPTION
**Pull request by big-andy-coates**
Fix for  [KAFKA-7064](https://issues.apache.org/jira/browse/KAFKA-7064)

I some how managed to get the Ids wrong when copying them from request.Resource to ConfigResource.Type, even though I double checked. :-/

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
cc @cmccabe

**Comment by ijuma** [comment]
Seems like the system test is still failing, but I don't know if it's the same reason:

> 16:41:59 test_id:    kafkatest.tests.client.client_compatibility_features_test.ClientCompatibilityFeaturesTest.run_compatibility_test.broker_version=0.11.0.2
> 16:41:59 status:     FAIL
> 16:41:59 run time:   58.891 seconds
> 16:41:59 
> 16:41:59 
> 16:41:59     
> 16:41:59 Traceback (most recent call last):
> 16:41:59   File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/venv/local/lib/python2.7/site-packages/ducktape-0.7.1-py2.7.egg/ducktape/tests/runner_client.py", line 132, in run
> 16:41:59     data = self.run_test()
> 16:41:59   File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/venv/local/lib/python2.7/site-packages/ducktape-0.7.1-py2.7.egg/ducktape/tests/runner_client.py", line 185, in run_test
> 16:41:59     return self.test_context.function(self.test)
> 16:41:59   File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/venv/local/lib/python2.7/site-packages/ducktape-0.7.1-py2.7.egg/ducktape/mark/_mark.py", line 324, in wrapper
> 16:41:59     return functools.partial(f, *args, **kwargs)(*w_args, **w_kwargs)
> 16:41:59   File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/tests/kafkatest/tests/client/client_compatibility_features_test.py", line 110, in run_compatibility_test
> 16:41:59     self.invoke_compatibility_program(features)
> 16:41:59   File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/tests/kafkatest/tests/client/client_compatibility_features_test.py", line 95, in invoke_compatibility_program
> 16:41:59     run_command(node, cmd, ssh_log_file)
> 16:41:59   File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/tests/kafkatest/tests/client/client_compatibility_features_test.py", line 50, in run_command
> 16:41:59     for line in node.account.ssh_capture(cmd):
> 16:41:59   File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/venv/local/lib/python2.7/site-packages/ducktape-0.7.1-py2.7.egg/ducktape/cluster/remoteaccount.py", line 614, in next
> 16:41:59     return next(self.iter_obj)
> 16:41:59   File "/home/jenkins/workspace/system-test-kafka-branch-builder/kafka/venv/local/lib/python2.7/site-packages/ducktape-0.7.1-py2.7.egg/ducktape/cluster/remoteaccount.py", line 300, in output_generator
> 16:41:59     raise RemoteCommandError(self, cmd, exit_status, stderr.read())
> 16:41:59 RemoteCommandError: ubuntu@worker1: Command '/opt/kafka-dev/bin/kafka-run-class.sh org.apache.kafka.tools.ClientCompatibilityTest --bootstrap-server worker4:9092,worker5:9092,worker6:9092 --num-cluster-nodes 3 --topic client_compat_features_topic_15293359551750924398 --offsets-for-times-supported True --describe-acls-supported True --create-topics-supported True --cluster-id-supported True --expect-record-too-large-exception False ' returned non-zero exit status 1.
> 16:41:59 

https://jenkins.confluent.io/view/All/job/system-test-kafka-branch-builder/1806/

**Comment by cmccabe** [comment]
Given that this is a last minute bug fix, can we make the change a bit smaller?  Maybe just include the enum value switch and the test for that?

**Comment by omkreddy** [comment]
ClientCompatabilityTest ```adminClient.describeAcls(AclBindingFilter.ANY).values().get()``` is failing against 0.11.0.2 broker.


```
java.lang.RuntimeException: Expected describeAclsSupported to be supported, but it wasn't.
	at org.apache.kafka.tools.ClientCompatibilityTest.tryFeature(ClientCompatibilityTest.java:503)
	at org.apache.kafka.tools.ClientCompatibilityTest.tryFeature(ClientCompatibilityTest.java:488)
	at org.apache.kafka.tools.ClientCompatibilityTest.testAdminClient(ClientCompatibilityTest.java:306)
	at org.apache.kafka.tools.ClientCompatibilityTest.run(ClientCompatibilityTest.java:226)
	at org.apache.kafka.tools.ClientCompatibilityTest.main(ClientCompatibilityTest.java:179)
Caused by: org.apache.kafka.common.errors.UnsupportedVersionException: Version 0 only supports literal resource pattern types
```

**Comment by cmccabe** [comment]
That unit test failure is catching a real bug (using `adminClient.describeAcls(AclBindingFilter.ANY).values().get()` shouldn't fail)

**Comment by big-andy-coates** [comment]
Hey @omkreddy, @cmccabe & @ijuma, couple of new changes pushed.  This should resolve all the issues I'm aware of.

System tests are running: https://jenkins.confluent.io/view/All/job/system-test-kafka-branch-builder/1812/

File: tests/kafkatest/tests/client/client_compatibility_features_test.py [hunk 1]
```diff
@@ -102,6 +102,8 @@ def invoke_compatibility_program(self, features):
     @parametrize(broker_version=str(LATEST_0_10_1))
     @parametrize(broker_version=str(LATEST_0_10_2))
     @parametrize(broker_version=str(LATEST_0_11_0))
+    @parametrize(broker_version=str(LATEST_1_0))
```
Comment (omkreddy) [inline tests/kafkatest/tests/client/client_compatibility_features_test.py:105]: we need to import LATEST_1_0, LATEST_1_1 version fields

Comment (big-andy-coates) [reply tests/kafkatest/tests/client/client_compatibility_features_test.py:105]: Doh!

Comment (big-andy-coates) [reply tests/kafkatest/tests/client/client_compatibility_features_test.py:105]: done.

**Comment by big-andy-coates** [comment]
System test passed.

Running again with newer versions included: https://jenkins.confluent.io/job/confluentinc/job/packaging/job/master/573/

**Review by junrao — APPROVED** [review]
@big-andy-coates : Thanks for the patch. LGTM

**Pull request merged**